src/RetryMiddleware.php

```diff
@@ -130,7 +130,7 @@ public function __invoke(
         ) {
             $this->updateHttpStats($value, $requestStats);
 
-            if ($value instanceof \Exception) {
+            if ($value instanceof \Exception || $this->isErrorObject($value)) {
```

Comment:
Why not just look for Throwable?

http://php.net/manual/en/class.throwable.php